Library/Homebrew/formula_creator.rb

```diff
@@ -31,24 +31,38 @@ def verify
       raise TapUnavailableError, @tap.name unless @tap.installed?
     end
 
-    def parse_url
-      path = Pathname.new(@url)
-      if @name.nil?
-        case @url
-        when %r{github\.com/(\S+)/(\S+)\.git}
-          @user = Regexp.last_match(1)
-          @name = Regexp.last_match(2)
-          @head = true
-          @github = true
-        when %r{github\.com/(\S+)/(\S+)/(archive|releases)/}
-          @user = Regexp.last_match(1)
-          @name = Regexp.last_match(2)
-          @github = true
-        else
-          @name = path.basename.to_s[/(.*?)[-_.]?#{Regexp.escape(path.version.to_s)}/, 1]
-        end
+    def self.name_from_url(url)
+      odebug "name_from_url: #{url}"
+      stem = Pathname.new(url).stem
+      odebug "stem: #{stem}"
+      # special cases first
+      name = if stem.start_with? "index.cgi"
+        # parsing gitweb URLs http://www.codesrc.com/gitweb/index.cgi?p=libzipper.git;a=summary
+        # added in https://github.com/Homebrew/brew/pull/9459
+        stem.rpartition("=").last
+      elsif url =~ %r{github\.com/\S+/(\S+)/(archive|releases)/}
+        Regexp.last_match(1)
+      else
+        pathver = Version.parse(stem).to_s
```

Comment:
Any comments that can be used for examples here?

Reply:
I believe it is covered by the http://digit-labs.org/files/tools/synscan/releases/synscan-5.02.tar.gz URL in spec. Here the version needs to be stripped from the `stem`.